src/message_private.h

```diff
@@ -107,11 +106,21 @@ typedef struct {
     qd_parsed_field_t   *ma_pf_to_override;
     qd_parsed_field_t   *ma_pf_trace;
     int                  ma_int_phase;
+    //qd_parsed_field_t   *parsed_message_annotations;
+
+    bool                 discard;                        // Should this message be discarded?
+    bool                 receive_complete;               // true if the message has been completely received, false otherwise
+    unsigned int         fanout;                         // The number of receivers for this message. This number does not include in-process subscribers.
 } qd_message_content_t;
 
 typedef struct {
     DEQ_LINKS(qd_message_t);   // Deque linkage that overlays the qd_message_t
-    qd_message_content_t *content;
+    qd_field_location_t   cursor;          // A pointer to the current location of the outgoing byte stream.
```

Comment:
qd_iterator_pointer_t might be more appropriate for a pointer into buffers. field_location has baggage that is not needed.